### ISSUE TITLE
accessToken is still null while a secured component is rendered

### ISSUE THREAD

**Issue by ddecrulle** [AxaGuilDEv/react-oidc#721]
### Issue and Steps to Reproduce

Hi, I made a CodeSandBox [here](https://codesandbox.io/s/intelligent-shape-k4co5y?file=/src/TestFetch.tsx) to reproduce and understand easily the bug. Just go on **Test Fetch** page without being authentificated.

The fact is that if a user go directly to a secure component without being authentificated, it is logically redirected to the authentication phase. Once the authentication is successful, it is redirected to the component but when the first rendering arrives, the token is still null. In the case where there are data fetches it is problematic because the first fetch leaves without a token.

This bug is **only visible on Firefox** where the service worker isn't used.

### Versions

4.X.X

### Screenshots

<img width="666" alt="Capture d’écran 2022-03-08 à 13 33 14" src="https://user-images.githubusercontent.com/81740200/157245373-4cdb35fe-3f62-49c1-ad07-09eeb64bfedc.png">

#### Expected

the first call goes out **with** a token

#### Actual

the first call goes out **without** a token

### Additional Details

I know that there is `fetchWithToken` and `withFetchToken` but it's difficult to use in case of hook api call with the possibility to disable auth.

**Comment by guillaume-chervet** [comment]
I think I can fix your problem. I have to investigate, but I know what append and where.

by waiting,

You may switch

```
 useEffect(() => {
      
    if (isLogged && count < 5) {
      const fetchUserInfoAsync = async () => {
        const res = await fetch("https://demo.identityserver.io/connect/userinfo", {
          headers: { Authorization: `Bearer ${accessToken}` }
        }
        )
        if (res.status != 200) {
          return null;
        }
        return res.json();
      };
      fetchUserInfoAsync().then((userInfo) => setUserInfo(userInfo))
      setCount(count + 1)
    }

  }, [accessToken, userInfo]);

```

By

```
 useEffect(() => {
      
    if (accessToken && count < 5) {
      const fetchUserInfoAsync = async () => {
        const res = await fetch("https://demo.identityserver.io/connect/userinfo", {
          headers: { Authorization: `Bearer ${accessToken}` }
        }
        )
        if (res.status != 200) {
          return null;
        }
        return res.json();
      };
      fetchUserInfoAsync().then((userInfo) => setUserInfo(userInfo))
      setCount(count + 1)
    }

  }, [accessToken, userInfo]);

```

**Comment by guillaume-chervet** [comment]
Thank you very much again for your feedback :)

**Comment by guillaume-chervet** [comment]
And also service worker sould work on mozilla, I have to fix a bug on it :)

**Comment by guillaume-chervet** [comment]
In v4 it is disabled by default.
a pullrequest will re-enable it, but still disabled by default.

(I have still some work and test to do)
https://github.com/AxaGuilDEv/react-oidc/pull/718

**Comment by guillaume-chervet** [comment]
https://github.com/AxaGuilDEv/react-oidc/pull/722 should correct the behavior !

It published under the version 4.2.11-alpha.0, does it resolve your need ?

**Comment by ddecrulle** [comment]
Thanks a lot for this quick fix ! It works for me. 

I've still some bug to discuss with you but didn't find pattern. 

For example sometimes I get stuck on blank page after login ([there](https://black-rock-0dc6b0d03.1.azurestaticapps.net/authentication/callback?code=C2B80268253F83323083135CE956CE624D469667485FD23C500CC11B9A7112D5&scope=openid%20profile%20email%20api%20offline_access&state=%2Fprofile&session_state=oGqEh5GUrMsEYTMTokRAf9qQIB3NmsylBjIdxIAYUd8.D4DB4CDEC803A02A318DD4D6DCA0691C) in you'r demo
